### ISSUE TITLE
Compact Machine Not setting location.

### ISSUE THREAD

**Issue by Badcholo** [thraaawn/CompactMachines#126]
So i set a compact machine. I upgrade it with a nether star. I go inside the machine. I set my /home. I come outside the machine. I break the upgraded machine. I /home and go inside the machine. I place the compact machine in a corner. If i right click it it teleports me to the center of the room. Works Great.

Issue. I place another upgraded machine in the same room. I go inside the machine. I try to come out and instead of placing me in the room it teleports me to overworld where i placed the first machine.
https://www.dropbox.com/s/z9rsh7if7j3bs9a/javaw%202015-05-22%2000-02-56-012.avi?dl=1

**Comment by thraaawn** [comment]
Yupp, you did not enter the first one via the PSD, but with /home. That is not how CMs are meant to work: Not a bug, but intended behavior. Also one of the reasons you get wither and nausea. Disabling that does not remove the original - not so obvious - limitations.

**Comment by Badcholo** [comment]
I did enter the first one with the PSD. I then upgraded it and removed it. Placing the second one should set the location to return to outside the block right? I have broke it and re-placed it but still doesn't return me to the outside of the block. I understand it is in the new dimension. ok thank you.

**Comment by Badcholo** [comment]
The issue isn't that it teleports me to overworld. The issue is the second compact machine that is placed within a compact machine. When i leave the second compact machine it doesn't teleport me into the 1rst compact machine even thou it was placed inside the first compact machine. i removed and placed it again but still doesn't teleport me into the first machine.

**Comment by thraaawn** [comment]
It does not matter where you placed a CM, it matters where you entered it from and how deep you are in. Entering a CM via /home does not increase your depth, but entering it via PSD does. Leaving one via PSD decreases the depth.
If your depth is 0 you will be teleported back to the overworld. Since your depth is 1 even though you are inside the inner CM, exiting it will mean your depth goes to 0, thus you are being teleported back to the overworld.

Bottom line: *do not enter CMs without a PSD*. They are not meant to work that way. You can disable the wither+nausea effects, but they are still not meant to work that way.

**Comment by Badcholo** [comment]
yes, i like the death thing. Since players are wondering why they have nausea and wither. Also a Comment that lets them know they were Crushed in the machine or to shrink themselves first before entering a machine. That would also remove the need for trying to want to set your bed inside the machine.

**Comment by Badcholo** [comment]
Any way to make trees or plants not grow inside the machine? I'd like to see players use CM for just machines, not for farming and breeding or mob grinders. Power sources, Ore Processing, Mass Production of Stuff, is enough of a great use for these machines.
